crates/hc_sandbox/src/ports.rs

```diff
@@ -46,19 +44,29 @@ pub async fn get_admin_ports(paths: Vec<PathBuf>) -> anyhow::Result<Vec<u16>> {
     Ok(ports)
 }
 
-pub(crate) async fn get_admin_api(port: u16) -> WebsocketResult<WebsocketSender> {
+pub(crate) async fn get_admin_api(
+    port: u16,
+) -> std::io::Result<(WebsocketSender, tokio::task::JoinHandle<()>)> {
     tracing::debug!(port);
-    websocket_client_by_port(port).await.map(|p| p.0)
+    websocket_client_by_port(port).await
 }
 
 async fn websocket_client_by_port(
     port: u16,
-) -> WebsocketResult<(WebsocketSender, WebsocketReceiver)> {
-    ws::connect(
-        url2!("ws://127.0.0.1:{}", port),
+) -> std::io::Result<(WebsocketSender, tokio::task::JoinHandle<()>)> {
+    let (send, mut recv) = ws::connect(
         Arc::new(WebsocketConfig::default()),
+        ([127, 0, 0, 1], port).into(),
     )
-    .await
+    .await?;
+    let task = tokio::task::spawn(async move {
+        while recv
+            .recv::<holochain_conductor_api::AdminResponse>()
+            .await
+            .is_ok()
+        {}
```

Comment:
The receiver is the thing receiving responses (and signals, but we don't care about signals in this case). The responses get forwarded to the sender that made the request, so we don't actually get them here. The `.is_ok()` is about successfully getting signals that we're ignoring. If there is an error, the sender side will also exit with an error, and that'll be more useful because we'll have made a request there.